plugins/module_utils/module_helper.py

```diff
@@ -155,27 +158,138 @@ def text(self):
         return self.msg or str(self.exc_val)
 
 
+class VarMeta(object):
+    def __init__(self, diff=False, output=False, change=None):
+        self.init = False
+        self.initial_value = None
+        self.value = None
+
+        self.diff = diff
+        self.change = diff if change is None else change
+        self.output = output
+
+    def set(self, diff=None, output=None, change=None):
+        if diff is not None:
+            self.diff = diff
+        if output is not None:
+            self.output = output
+        if change is not None:
+            self.change = change
+
+    def set_value(self, value):
+        if not self.init:
+            self.initial_value = value
+            self.init = True
+        self.value = value
+        return self
+
+    @property
+    def has_changed(self):
+        return self.change and (self.initial_value != self.value)
+
+    @property
+    def diff_result(self):
+        return None if not (self.diff and self.has_changed) else {
+            'before': self.initial_value,
+            'after': self.value,
+        }
+
+    def __str__(self):
+        return"<VarMeta: value={0}, initial={1}, diff={2}, output={3}, change={4}>".format(
```

Comment:
I'm a bit surprised that none of the linters was screaming at this one :)
```suggestion
        return "<VarMeta: value={0}, initial={1}, diff={2}, output={3}, change={4}>".format(
```